### PR TITLE
implement stats/qps handler

### PR DESCRIPTION
**Pull request by veshij**
https://github.com/ns1/ns1-go/issues/39

### CODE REVIEW

**Comment by mburtless** [comment]
Thanks for the contribution, this is looking great!  I should be able to review this and provide some feedback for you shortly.

File: rest/stat.go [hunk 1]
```diff
@@ -1,15 +1,61 @@
...
+
+	var r map[string]float32
+	resp, err := s.client.Do(req, &r)
+	if err != nil {
+		switch err.(type) {
+		case *Error:
+			switch err.(*Error).Message {
+			case "zone not found":
+				return 0, ErrZoneMissing
+			}
```
Comment (mburtless) [inline rest/stat.go:53]: We should handle the case where GetRecordQps query returns `record not found` error and return `ErrRecordMissing`

File: rest/stat.go [hunk 2]
```diff
@@ -1,15 +1,61 @@
...
+
+// Returns current queries per second (QPS) for a specific record.
+// The QPS number is lagged by approximately 30 seconds for statistics collection;
+// and the rate is computed over the preceding minute.
+func (s StatsService) GetRecordQps(zone, record, t string) (float32, error) {
+	path := fmt.Sprintf("%s/%s/%s/%s", statsQpsEndpoint, zone, record, t)
+	return s.getQps(path)
+}
+
+func (s StatsService) getQps(path string) (float32, error) {
```
Comment (mburtless) [inline rest/stat.go]: Other handler Get functions also return a pointer to the http.Response.  Is there a reason we wouldn't want to do that in Stats?

**Review by mburtless — CHANGES_REQUESTED** [review]
Looking good, just a couple of tiny issues to fix.  Would also be good to get a small example added for this handler.  Would you mind adding one to `_examples` when you get a chance?

**Comment by mburtless** [comment]
👋 Hi @veshij!  Just wanted to circle back on this.  Will you be able to address the points from the review?  Thanks!

**Comment by veshij** [comment]
> 👋 Hi @veshij! Just wanted to circle back on this. Will you be able to address the points from the review? Thanks!

oops, thanks for pinging me, working on comments right now.

File: rest/stat.go (hunk 2 above)
Comment (veshij) [reply rest/stat.go]: I used commented-out code as a reference: `GetQPSStats() (v float64, err error)`.
Updated to return qps and *http.Response (similar to https://github.com/ns1/ns1-go/blob/03b09e404c10ef6d966b038f97364710db4f972f/rest/account_setting.go)

File: rest/stat.go (hunk 1 above)
Comment (veshij) [reply rest/stat.go:53]: Added check (should be implemented in more generic way though)

**Comment by veshij** [comment]
@mburtless 
- added example
- updated `Stats.Get*()` signatures to return *http.Response as well
- added check for `ErrRecordMissing`

**Comment by mburtless** [comment]
Thanks for the update @veshij, this is looking great.  Travis noted a few linting issues, once these get resolved I think this should be good to go.

```
rest/stat.go:9:7: const statsQpsEndpoint should be statsQPSEndpoint
rest/stat.go:14:1: comment on exported method StatsService.GetQps should be of the form "GetQps ..."
rest/stat.go:17:24: method GetQps should be GetQPS
rest/stat.go:21:1: comment on exported method StatsService.GetZoneQps should be of the form "GetZoneQps ..."
rest/stat.go:24:24: method GetZoneQps should be GetZoneQPS
rest/stat.go:29:1: comment on exported method StatsService.GetRecordQps should be of the form "GetRecordQps ..."
rest/stat.go:32:24: method GetRecordQps should be GetRecordQPS
rest/stat.go:37:24: method getQps should be getQPS
rest/stat.go:60:18: should replace errors.New(fmt.Sprintf(...)) with fmt.Errorf(...)
```

**Comment by veshij** [comment]
> Thanks for the update @veshij, this is looking great. Travis noted a few linting issues, once these get resolved I think this should be good to go.
> 
> ```
> rest/stat.go:9:7: const statsQpsEndpoint should be statsQPSEndpoint
> rest/stat.go:14:1: comment on exported method StatsService.GetQps should be of the form "GetQps ..."
> rest/stat.go:17:24: method GetQps should be GetQPS
> rest/stat.go:21:1: comment on exported method StatsService.GetZoneQps should be of the form "GetZoneQps ..."
> rest/stat.go:24:24: method GetZoneQps should be GetZoneQPS
> rest/stat.go:29:1: comment on exported method StatsService.GetRecordQps should be of the form "GetRecordQps ..."
> rest/stat.go:32:24: method GetRecordQps should be GetRecordQPS
> rest/stat.go:37:24: method getQps should be getQPS
> rest/stat.go:60:18: should replace errors.New(fmt.Sprintf(...)) with fmt.Errorf(...)
> ```

updated, Travis is happy now.

**Comment by veshij** [comment]
> updated, Travis is happy now.
@mburtless could you take a look please?

**Review by mburtless — APPROVED** [review]

**Comment by mburtless** [comment]
👍 Looks good, thanks again for the contribution!  This should get merged on Monday (we don't release on Fridays).

**Comment by veshij** [comment]
> 👍 Looks good, thanks again for the contribution! This should get merged on Monday (we don't release on Fridays).

gentle ping :)

**Pull request merged**